releasenotes/notes/42119.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: release-notes/v2
+
+# This YAML file describes the format for specifying a release notes entry for Istio.
+# This should be filled in for all user facing changes.
+
+# kind describes the type of change that this represents.
+# Valid Values are:
+# - bug-fix -- Used to specify that this change represents a bug fix.
+# - security-fix -- Used to specify that this change represents a vulnerability fix.
+# - feature -- Used to specify a new feature that has been added.
+# - test -- Used to describe additional testing added. This file is optional for
+#   tests, but included for completeness.
+kind: bug-fix
+
+# area describes the area that this change affects.
+# Valid values are:
+# - traffic-management
+# - security
+# - telemetry
+# - installation
+# - istioctl
+# - documentation
+area: traffic-management
+
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - https://github.com/istio/istio/issues/42094
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+  - |
+    **Added** partial wildcard checks to `Gateway` and `Sidecar`.
```

Comment:
Thanks for adding the RN, but I'm not sure that this adequately tells the user what this PR does. Even looking at the referenced issue, it is hard to say what these additional checks do.

Reply:
> '*-mr.api.myservice.com'

envoy does not allow this as server_names.

Reply:
I really don't know how to write RN. Do you have any good suggestions?

Reply:
Maybe something like: **Added** a check for partial wildcards to `Gateway` and `Sidecar` as they are not supported by Envoy.

Reply:
Thank you very much.  @ericvn

Reply:
Done